src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,12 +1,32 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from '../Loader';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
+import { getUser } from '../../api';
+
+type Props = {
+  visibleModal: Todo,
+  setVisibleModal: (todo: Todo | null) => void;
+};
+
+export const TodoModal: React.FC<Props> = ({
+  visibleModal, setVisibleModal,
+}) => {
+  const [user, setUser] = useState<User | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    getUser(visibleModal.userId)
+      .then(setUser)
+      .catch(error => new Error(error.message))
+      .finally(() => setLoading(false));
+  }, [visibleModal.userId]);
```

Comment:
The useEffect is dependent on 'visibleModal.userId' but not 'setVisibleModal', which is also used inside the effect. It might cause problems if 'setVisibleModal' changes and the effect doesn't re-run. Include all used state variables in dependency array or use the callback form of the state setter function inside the effect